### ISSUE TITLE
The "new" releases completely broken, please update release notes

### ISSUE THREAD

**Issue by karniemi** [robotframework/jrobotremoteserver#67]
The new releases from org.robotframework are completely broken, please update release notes to reflect this.

We tried upgrading from 3.0 to 4.1.0, but it was a complete failurewith errors like these:
* ` java.lang.ClassCastException: java.util.ArrayList cannot be cast to java.lang.String`
  * java method:` public void doStuff(String text, List<String> skip_txts) {}`
  * robot call: `do stuff	${LANG_PREFIX_${lang}}	${NO_TRANS}`
  * robot vars: `@{NO_TRANS}=	__ 	default	word1	word2`
* also, we can see remote KWs being called but not actually doing anything, but still returning success.
  * robot vars: `&{VEHICLE}	TYPE=Other	REGISTERNUMBER=${REGISTERPLATE}`
  * robot call: `Check vehicle	$VEHICLE`
  * java:
```
public void checkVehicle(Map<String,String> vehicleDetails) throws IOException {
		checkVehicle(vehicleDetails, false);
	}

public void checkVehicle(Map<String,String> vehicleDetails, boolean click) throws IOException {
    <clip>
    vehicleDetails.get("REGISTERNUMBER") // RETURNS NULL !!!!
}
```

It seems that at least passing any arrays and maps is just completely broken.

**Comment by karniemi** [comment]
BTW, we are still on Robot Framework 3. I could not find any compatibility matrix for jrobotremoteserver, is there any?

**Comment by karniemi** [comment]
@Hi-Fi ...any idea about these issues?

**Comment by Hi-Fi** [comment]
I think that release is tested when done, and it has working. But it's done on Java 8, so there might be issues with other versions.

There shouldn't be differences on RF versions itself, as new version just adds couple more API expectation and if those are not available usage is just ignored.

So, for issue: What Java and Maven version you're using?

**Comment by karniemi** [comment]
We are using maven-3.6.3, but only to compile & package our remote-library - we are not running robot-tests using maven. We are running the remote-library in jre8.

**Comment by karniemi** [comment]
To me it would seem that the interface between robot-framework and jrobotremoteserver is now broken for passing (at least) arrays, lists and maps. These did work in jrobotremoteserver:3.0.

**Comment by Hi-Fi** [comment]
There were quite a lot of changes between 3 and 4, one of the biggest being change of javalib-core.

But I wouldn't say "completely broken", as there are tests that are run on each release, and those pass: https://github.com/robotframework/jrobotremoteserver/tree/develop/atest/tests.

I know that there's issue with some more complex values like objects (see #55), but those basic values should work fine.

I haven't used this for a while, so there might be something outdated. Also note that with #68 build was changed to be made with Java 11 (dropping Java 8 support). So next release would be probably 5.x.

**Comment by karniemi** [comment]
Well, our test suites depend on using complex values - and the new versions of jrobotremoteserver are so badly broken ( at least in that area ) that we can't use the new versions.

**Comment by Hi-Fi** [comment]
I'd say that way you put things doesn't help to get things fixed. I haven't myself used jrobotremotelibrary for ages, so I don't have tests where fixes would be needed. 

I think this issue could be closed as this is then duplicate to #55. Which would have probably got more attention if stating that issue is with complex objects and not just "everything broken".
